spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpoint.java

```diff
@@ -138,8 +140,17 @@ public void processPacket(final Stanza packet) {
 				Object messageBody = xmppMessage.getBody();
 
 				if (ChatMessageListeningEndpoint.this.payloadExpression != null) {
+					EvaluationContext evaluationContext = ChatMessageListeningEndpoint.this.evaluationContext;
+
+					List<ExtensionElement> extensions = xmppMessage.getExtensions();
+					if (extensions.size() == 1) {
+						ExtensionElement extension = extensions.get(0);
+						evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+						evaluationContext.setVariable("extension", extension);
+					}
```

Comment:
Why limit to 1 extension ?

Maybe `#extension` when there's one and `#extensions` when > 1 ??